src/DataTomeUtils.h

```diff
@@ -21,8 +25,8 @@ void swap(TypeOfArray &a, TypeOfArray &b) {
   b = temp;
 }
 
-template <typename T>
-const T &dt_min(const T &a, const T &b) {
+template <typename TypeOfArray>
+const TypeOfArray &dt_min(const TypeOfArray &a, const TypeOfArray &b) {
```

Comment:
Nice change to comply with the standard used on the codebase! I was thinking of writing it myself. Thank you for noticing it.